structure_tests/drivers/docker_driver.go

```diff
@@ -0,0 +1,357 @@
+// Copyright 2017 Google Inc. All rights reserved.
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+
+//     http://www.apache.org/licenses/LICENSE-2.0
+
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package drivers
+
+import (
+	"bytes"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"strings"
+	"testing"
+	"time"
+
+	"context"
+
+	"bufio"
+
+	"regexp"
+
+	"github.com/GoogleCloudPlatform/runtimes-common/structure_tests/types/unversioned"
+	"github.com/fsouza/go-dockerclient"
+)
+
+type DockerDriver struct {
+	cli docker.Client
+}
+
+var originalImage, currentImage string
+
+func (d DockerDriver) Info() string {
+	return fmt.Sprintf("DockerDriver:\nOriginalImage: %s\ncurrentImage: %s\ncli: %s\n",
+		originalImage, currentImage, d.cli)
+}
+
+func NewDockerDriver(image string) DockerDriver {
+	newCli, err := docker.NewClientFromEnv()
+	if err != nil {
+		panic(err)
+	}
+	originalImage = image
+	currentImage = image
+	return DockerDriver{
+		cli: *newCli,
+	}
+}
+
+func (d DockerDriver) ProcessCommand(t *testing.T, envVars []unversioned.EnvVar, fullCommand []string,
+	shellMode bool, checkOutput bool) (string, string, int) {
+
+	if len(fullCommand) == 0 {
+		t.Logf("empty command provided: skipping...")
+		return "", "", -1
+	}
+	var env []string
+	for _, envVar := range envVars {
+		env = append(env, envVar.Key+"="+envVar.Value)
+	}
+	stdout, stderr, exitCode := d.exec(t, env, fullCommand)
+
+	if stdout != "" {
+		t.Logf("stdout: %s", stdout)
+	}
+	if stderr != "" {
+		t.Logf("stderr: %s", stderr)
+	}
+	//reset image for next test
+	currentImage = originalImage
+	return stdout, stderr, exitCode
+}
+
+func (d DockerDriver) SetEnvVars(t *testing.T, vars []unversioned.EnvVar) []unversioned.EnvVar {
+	// currently, this does not support substitution (e.g. PATH=/env/bin:$PATH), so setting env vars
+	// is broken. need to find a way to retrieve the existing values from the container before setting
+	if len(vars) == 0 {
+		return nil
+	}
+	ctx := context.Background()
+	container, err := d.cli.CreateContainer(docker.CreateContainerOptions{
+		Config: &docker.Config{
+			Image:        currentImage,
+			Cmd:          []string{"NOOP_COMMAND_DO_NOT_RUN"},
+			AttachStdout: true,
+			AttachStderr: true,
+		},
+		HostConfig:       nil,
+		NetworkingConfig: nil,
+		Context:          ctx,
+	})
+	if err != nil {
+		t.Errorf("Error creating container: %s", err.Error())
+		return nil
+	}
+	image, err := d.cli.InspectContainer(container.ID)
```

Comment:
Can you use InspectImage instead of needing to create a container?
https://github.com/fsouza/go-dockerclient/blob/master/image.go#L200

Reply:
ah yep, didn't see that `Image` has a `Config` as well. updated